crates/api/api/src/community/tag.rs

```diff
@@ -1,61 +1,55 @@
 use activitypub_federation::config::Data;
 use actix_web::web::Json;
 use chrono::Utc;
-use lemmy_api_utils::{context::LemmyContext, utils::check_community_mod_action};
+use lemmy_api_utils::{
+  context::LemmyContext,
+  send_activity::{ActivityChannel, SendActivityData},
+  utils::check_community_mod_action,
+};
 use lemmy_db_schema::{
   source::{
     community::Community,
     tag::{Tag, TagInsertForm, TagUpdateForm},
   },
   traits::Crud,
 };
-use lemmy_db_views_community::api::{CreateCommunityTag, DeleteCommunityTag, UpdateCommunityTag};
+use lemmy_db_views_community::api::{CreateCommunityTag, DeleteCommunityTag};
 use lemmy_db_views_local_user::LocalUserView;
-use lemmy_utils::{error::LemmyResult, utils::validation::tag_name_length_check};
+use lemmy_db_views_site::SiteView;
+use lemmy_utils::{error::LemmyResult, utils::validation::is_valid_actor_name};
+use url::Url;
 
 pub async fn create_community_tag(
   data: Json<CreateCommunityTag>,
   context: Data<LemmyContext>,
   local_user_view: LocalUserView,
 ) -> LemmyResult<Json<Tag>> {
+  // reuse this existing function for validation
+  let site_view = SiteView::read_local(&mut context.pool()).await?;
+  let local_site = site_view.local_site;
+  is_valid_actor_name(&data.name, local_site.actor_name_max_length)?;
```

Comment:
Should we allow uppercase letters in tags? Might be confusing to let tags have different cases.

Reply:
It doesnt have to be very restrictive as mods can manage the tags in a way to avoid confusion.